code/mission/missioncampaign.cpp

```diff
@@ -1079,10 +1083,59 @@ void mission_campaign_store_variables(int persistence_type, bool store_red_alert
 	}
 }
 
+// jg18 - adapted from mission_campaign_store_variables()
+void mission_campaign_store_containers(ContainerType persistence_type, bool store_red_alert)
+{
+	if (!sexp_container_has_persistent_non_eternal_containers()) {
+		// nothing to do
+		return;
+	}
+
+	if (store_red_alert) {
+		Campaign.persistent_containers = Campaign.red_alert_containers;
+	}
+
+	for (const auto &container : get_all_sexp_containers()) {
+		if (!container.is_eternal()) {
+			if (any(container.type & persistence_type)) {
+				// see if we already have a container with this name
+				auto cpc_it = std::find_if(Campaign.persistent_containers.begin(),
+					Campaign.persistent_containers.end(),
+					[container](const sexp_container& cpc) {
+						return !stricmp(container.container_name.c_str(), cpc.container_name.c_str());
```

Comment:
Should this search also take container type comparison into account, in addition to name comparison? Or was that type checking already done in `parse_sexp_containers()`?